Events/Christmas 201X.lua

```diff
@@ -263,96 +263,6 @@ end
 local getNeededXSpeed = function(distance)
 	-- Returns the needed xSpeed to cover distance with the given friction over a 0° ground.
 	return math.floor(distance ^ .5 + 0.5)
```

Comment:
Explanation:
	V_0 = initial velocity (px/f (pixels / frame))
	V_f = final velocity (px/f)
	d = distance (px (pixels))
	a = acceleration (px/f² (pixels / frame²))
	t = time (frame)
	d = V_0 * t + 1/2 * a * t²
	a = (V_f - V_0) / t
	Since we've got a friction, there is a counter force applied over the object that we're
	trying to move, and hence, a counter acceleration. We do know distance, final velocity
	(which should be 0), but we don't know the time, acceleration nor initial velocity.
	We've got only two equations with 3 unknown values!
	Doing some tests, we can get the time it takes and the acceleration it has like this:
	V_0 = some value (1, 2, 3, 4...)
	V_f = 0
	d = total distance after the test
	a = ?
	t = ?
	d = V_0 * t + 1/2 * (-V_0 / t) * t²
	d = V_0 * t - 1/2 * V_0 * t
	d = 1/2 * V_0 * t
	(2 * d) / V_0 = t
	a = -V_0 / t
	Note that these formulas are from a constant accelerated/deceletared movement.
	A friction does not apply a constant acceleration/deceleration, but this seems to work
	pretty nice. Acceleration seems to converge to about -0.115 with a ground friction of 0.3
	and a fish object (id 6300). Our event cases are not a huge list of cases, so this will
	work like a charm. So then, we've got acceleration value. We've got only two
	unknown values: time and initial speed!
	V_0 = ? px/f
	V_f = 0 px/f
	d = N px
	a = -0.115 px/f²
	t = ? f
	d = V_0 * t + 1/2 * a * t²
	a = (V_f - V_0) / t
	Both equations have two unknown values, we can use a equation moving an unknown value
	to a free side, and replacing that value with the equivalent. See:
	We could use the second equation like this:
	t = (V_f - V_0) / a
	and then replace every time variable on the first equation with this value
	d = V_0 * ((V_f - V_0) / a) + 1/2 * a * ((V_f - V_0) / a)²
	See? Only one unknown value: V_0.
	Clearing a bit the equation it gives us this:
	d = (V_f * V_0 - V_0²) / a + (V_f - V_0)² / (2 * a)
	It is a simple fraction addition. We can easily simplify it!
	d = (2 * (V_f * V_0 - V_0²) + (V_f - V_0)²) / (2 * a)
	Apply distributive property and it gives us this:
	d = (2 * V_f * V_0 - 2 * V_0² + (V_f - V_0) * (V_f - V_0)) / (2 * a)
	Keep applying distributive property on (V_f - V_0) * (V_f - V_0).
	It gives us V_f² - V_f * V_0 - V_0 * V_f + V_0². simplify it: V_f² - 2 * V_f * V_0 + V_0²
	d = (2 * V_f * V_0 - 2 * V_0² + V_f² - 2 * V_f * V_0 + V_0²) / (2 * a)
	There is a positive 2 * V_f * V_0 and a negative one. We can delete them.
	There are two negative V_0² and a positive one. We can leave a single negative one.
	d = (-V_0² + V_f²) / (2 * a)
	Multiply both sides by 2 * a:
	2 * d * a = -V_0² + V_f²
	Substract V_f² to both sides:
	2 * d * a - V_f² = -V_0²
	And then make it a square root.
	±sqrt(-2 * d * a - V_f²) = V_0
	It can be either positive or negative. Now, we know that V_f is 0. We can remove it from our equation:
	±sqrt(-2 * d * a) = V_0
	Know that -2 and a being negative gives us a positive value.
	Our units are px for distance, px/f for velocity and px/f² for acceleration. The -2 doesn't have any unit.
	sqrt(px * px/f²) gives us sqrt(px²/f²). We apply distributive property to sqrt(px²)/sqrt(f²) and then cancel
	squares with roots. It gives us px/f. A velocity unit. Seems like we're going for the right way.
	(-2 * d * -0.115) ^ .5 would do the job. We can also remove the - symbols and turn constants into a single value.
	So, since our formula is not 100% corrent and has a fail rate, we need to round the value!
	math.floor((d * 0.23) ^ .5 + .5) is all that we need.
	On the event, this speed seems to be a bit low because of ground rotations.
	We can fix that by just setting the constant to a bit more. 1 is a good value, so
	math.floor(d ^ .5 + .5) works just nice!